stardog/admin.py

```diff
@@ -1226,3 +1331,118 @@ def get_options(self):
 
     def __repr__(self):
         return self.name
+
+class Cache(object):
+    """Cached data
+
+    A cached dataset from a query or named/virtual graph.
+
+    See Also:
+        https://www.stardog.com/docs/#_cache_management
+    """
+    @classmethod
+    def __new_cache(
+            cls,
+            client,
+            name,
+            target,
+            database=None,
+            refresh_script=None,
+            query=None,
+            graph=None):
+        params = {
+            'name': name,
+            'target': target,
+            'database': database,
+            'refreshScript': refresh_cript,
+            'query': query,
+            'graph': graph,
+        }
+        client.post('/admin/cache', json=params)
+        return Cache(name, client)
+
+    def __init__(self, name, client):
+        """Initializes a new cached dataset from a query or named/virtual graph.
+
+        Use :meth:`stardog.admin.Admin.new_cached_graph` or
+        :meth:`stardog.admin.Admin.new_cached_query` instead of
+        constructing manually.
+        """
+        self.name = name
+        self.client = client
+        self.status()  # raises exception if cache doesn't exist on the server
+
+    def drop(self):
+        """Drops the cache."""
+        self.delete('/admin/cache/{}'.format(self.name))
+
+    def refresh(self):
+        """Refreshes the cache."""
+        self.client.post('/admin/cache/refresh/{}'.format(self.name))
+
+    def status(self):
+        """Retrieves the status of the cache."""
+        r = self.client.post('/admin/cache/status', json=[self.name])
+        return r.json()
+
+    def __repr__(self):
+        return self.name
+
+
+class CacheTarget(object):
+    """Cache Target Server
+    """
+    def __init__(self, name, client):
+        """Initializes a cache target.
+
+        Use :meth:`stardog.admin.Admin.new_cache_target` instead of
+        constructing manually.
+        """
+        self.cache_target_name = name
+        self.path = '/admin/cache/target/{}'.format(name)
+        self.client = client
+        self.details = {}
+        self.refresh = self.__refresh()
+
+    def __refresh():
+        target = []
+        r = self.client.get('/admin/cache/target')
+        self.targets = r.json()
+        target = next((t for t in r.json() if t['name'] == self.name), {})
+        self.details.update(target)
+
+    @property
+    def name(self):
+        """The name (URI) of the cache target."""
+        return self.cache_target_name
+
+    @property
+    def hostname(self):
+        """The hostname of the cache target."""
+        return self.details['hostname']
+
+    @property
+    def port(self):
+        """The port of the cache target."""
+        return self.details['port']
+
+    @property
+    def username(self):
+        """The username of the cache target."""
+        return self.details['username']
+
+    @property
+    def password(self):
```

Comment:
I am not really familiar with cache targets, but I don't see this documented in the API docs. It sounds to me that if a user creates a cache target, another user could load that cache target and do `print(cachetarget.password)` and it would return a plaintext password. Is that correct? if so, could this considered a security vulnerability? or is this by design?